src/pages/new-chat/new-chat.ts

```diff
@@ -9,22 +12,30 @@ import { User } from 'api/models/whatsapp-models';
   templateUrl: 'new-chat.html'
 })
 export class NewChatComponent implements OnInit {
-  users;
+  searchPattern = '';
   senderId: string;
+  users: Observable<User[]>;
 
   constructor(
-    public navCtrl: NavController,
-    public viewCtrl: ViewController,
-    public alertCtrl: AlertController
+    private navCtrl: NavController,
+    private viewCtrl: ViewController,
+    private alertCtrl: AlertController
   ) {
     this.senderId = Meteor.userId();
   }
 
   ngOnInit() {
-    MeteorObservable.subscribe('users').subscribe(() => {
-      MeteorObservable.autorun().subscribe(() => {
-        this.users = this.findUsers().zone();
-      });
+    this.findContacts().then((contacts) => {
+      // Pluck all contacts' phone numbers into a flat array
+      contacts = contacts && _.chain(contacts)
+        .pluck('phoneNumbers')
```

Comment:
After checking this code snippet, I think it wouldn't make sense to use a `getDeep` handler here since we flatten the array after plucking the phone numbers, a solution which is not provided by a `getDeep` handler.